rust/nasl-interpreter/src/built_in_functions/ssh.rs

```diff
@@ -0,0 +1,2107 @@
+                                if !p.echo {
+                                    answers.push(password.unwrap_or_default().to_string());
+                                } else {
+                                    answers.push(String::new());
+                                };
+                            }
+                            match session
+                                .session
+                                .userauth_keyboard_interactive_set_answers(&answers)
+                            {
+                                Ok(_) => {
+                                    return Ok(NaslValue::Number(0));
+                                }
+                                Err(_) => break,
+                            }
+                        }
+                        Ok(_) => {
+                            if verbose {
+                                ctx.logger()
+                                    .info(&format!(
+                                    "SSH keyboard-interactive authentication failed for session {}",
+                                    session_id));
+                            };
+                            continue;
+                        }
+                        Err(_) => {
+                            return Err(FunctionErrorKind::Diagnostic(
+                                format!(
+                                    "Failed setting user authentication for SessionID {}",
+                                    session_id
+                                ),
+                                Some(NaslValue::Null),
+                            ));
+                        }
+                    };
+                }
+            };
+
+            // If we have a private key, try public key authentication.
+            if privatekey.is_none() && methods.contains(AuthMethods::PUBLIC_KEY) {
+                match SshKey::from_privkey_base64(privatekey.unwrap_or_default(), passphrase) {
+                    Ok(k) => match session.session.userauth_try_publickey(None, &k) {
+                        Ok(AuthStatus::Success) => {
+                            match session.session.userauth_publickey(None, &k) {
+                                Ok(AuthStatus::Success) => {
+                                    return Ok(NaslValue::Number(0));
+                                }
+                                _ => {
+                                    if verbose {
+                                        ctx.logger().info(&format!("SSH authentication failed for session {}: No more authentication methods to try", session_id));
+                                    };
+                                }
+                            }
+                        }
+                        _ => {
+                            if verbose {
+                                ctx.logger().info(&format!("SSH public key authentication failed for session {}: Server does not want our key", session_id));
```

Comment:
because this is use an ENV variable to switch it on, different from the general debug level. It will only be available if it is set via, which also turns the libssh debug log on     
```
let verbose = env::var("OPENVAS_LIBSSH_DEBUG")